routes/decoratorQueryArgs.go

```diff
@@ -61,11 +61,13 @@ type (
 	}
 
 	// QueryArg defines an argument by its name and its type. A description
-	// can be used for documentation purposes.
+	// can be used for documentation purposes. The argument will be mandatory
+	// if you notice it.
 	QueryArg struct {
 		Name        string
 		Description string
 		Type        QueryParser
+		Mandatory   bool
```

Comment:
Previous behavior was : every `QueryArg` represents a mandatory query argument. Now you add a `Mandatory` field whose default value is `false` (the zero bool value), so all previous uses of `QueryArg` now become optional. Finally, you _did not_ go back on all previous definitions of `QueryArg` to make them mandatory again. You should be very careful with this kind of changes and think about how you change the behavior of what’s already there. An easy fix is to replace your `Mandatory` field by an `Optional` field, which defaulting to `false`, _does not_ change the behavior of previously defined `QueryArgs`. Please do that.
The other solution would be to go back on every definition of a `QueryArg` to add `true` to its definition, which is practical in this situation because the codebase is young, but would be harder with a larger codebase.